system-helper/flatpak-system-helper.c

```diff
@@ -117,9 +117,16 @@ schedule_idle_callback (void)
 }
 
 static FlatpakDir *
-dir_get_system (void)
+dir_get_system (const char *installation)
 {
-  FlatpakDir *system = flatpak_dir_get_system_default ();
+  FlatpakDir *system = NULL;
+
+  if (installation != NULL && *installation != '\0')
+    system = flatpak_dir_get_system_by_id (installation);
+
+  if (system == NULL)
+    system = flatpak_dir_get_system_default ();
+
```

Comment:
I don't think this is right. If you pass in a random installation name it should not fall back to the default.
Instead it should report an error.